### DISCUSSION TITLE
[REQUEST] Increase the child entity movement mass limit

### DISCUSSION THREAD

**Discussion by cloudrambler** [tsunamayo/Starship-EVO#4035] [Suggestions]
I am building some large turrets. They are not stupidly large, just a reasonable proportion for the large weapons. These are ending up around 150% to 200% of the child entity movement mass limit. Right now the turret moves anyway (not resetting position), but I am assuming that the turrets will be harmonised with standard mechanisms and have consistent mass limits. So in the future they may not work.

If turrets will always be infinite mass, that would be really useful to know. In which case it would also be useful to make them reset at any mass. But if they will have a fixed mass, I would like to request that it be larger. Personally I would like 100Tons.

I also have ideas for cool ships with moving thruster pods and those could be a few hundred Tons. Can we have larger/scalable actuators or the ability to link actuators together? I'm sure this is not a new request, but it is limiting at the moment, especially for turrets/engine pods that are very dense and so don't contain large block counts.

**Comment by ProPeach** [comment]
Ideally, I think a vertically stretchable rotor would be great to increase mass limits. There was some good talk on this a long while ago in issue #2000 
I also think that when when a turret hits its max limit, it should keep moving but very, very slowly.
So the mass/rotation curve on a rotor could look like this- (sorry for the terrible writing :D )
![image](https://user-images.githubusercontent.com/27857711/105520774-0663f100-5cd3-11eb-9afd-772edc704445.png)
The equation for this graph is y=1000e^(-0.002x)+50 if you want to play around with it. To be clear this is just an example to get the idea across, not a suggestion for the exact curve.

Exponential decay on the speed, but with a minimum so that no turret is completely useless. This minimum should scale with the power of a rotor, so an overloaded 1m tall rotor would be far slower than an overloaded 2m tall rotor for example.


This would make turret building a lot more interesting, and offer more powerful turrets a natural nerf to agility ("Their ships are so small they're evading our turbo lasers!")
